optuna/visualization/intermediate_values.py

```diff
@@ -25,13 +25,27 @@ def plot_intermediate_values(study):
 
             import optuna
 
+            # Derivative function for x**2
+            def df(x):
+                return 2*x
+
             def objective(trial):
-                x = trial.suggest_uniform('x', -100, 100)
-                y = trial.suggest_categorical('y', [-1, 0, 1])
-                return x ** 2 + y
+
+                next_x = 1  # We start the search at x=1
+                gamma = trial.suggest_loguniform('alpha', 1e-5, 1e-1)  # Step size multiplier
+
+                # Stepping through gradient descent to find the minima of x**2
+                for step in range(100):
+                    current_x = next_x
+                    next_x = current_x - gamma * df(current_x)
+
+                    delta = next_x - current_x
```

Comment:
How about defining the function "f", i.e. `x ** 2` and report/return the value of "f" given the current x?